MotionMark/tests/resources/main.js

```diff
@@ -155,6 +155,29 @@ Controller = Utilities.createClass(
         return averageFrameLength;
     },
 
+    _getFrameType: function(samples, i)
+    {
+        return samples[sampleTypeIndex][i];
+    },
+
+    _getComplexity: function(samples, i)
+    {
+        return samples[sampleComplexityIndex][i];
+    },
+
+    _getFrameLength: function(samples, i)
+    {
+        return samples[sampleTimeIndex][i] - samples[sampleTimeIndex][i - 1];
+    },
+    
+    _previousFrameComplexity: function(samples, currentSampleCount)
+    {
+        if (currentSampleCount > 0)
+            return this._getComplexity(samples, currentSampleCount - 1);
+
+        return 0;
```

Comment:
Maybe it makes more sense if we return `this._getComplexity(0)` instead of 0 if `currentSampleCount` is equal to 0.

```
return this._getComplexity(samples, Math.max(currentSampleCount - 1, 0));
```